packages/client/src/schema/repository.ts

```diff
@@ -453,12 +458,19 @@ export class RestRepository<Data extends BaseData, Record extends XataRecord = D
     });
   }
 
-  async search(query: string, options: { fuzziness?: number } = {}): Promise<SelectedPick<Record, ['*']>[]> {
+  async search(
+    query: string,
+    options: { fuzziness?: number; filter?: Filter<Record> } = {}
+  ): Promise<SelectedPick<Record, ['*']>[]> {
     const fetchProps = await this.#getFetchProps();
 
-    const { records } = await searchBranch({
-      pathParams: { workspace: '{workspaceId}', dbBranchName: '{dbBranch}' },
-      body: { tables: [this.#table], query, fuzziness: options.fuzziness },
+    const { records } = await searchTable({
+      pathParams: { workspace: '{workspaceId}', dbBranchName: '{dbBranch}', tableName: this.#table },
+      body: {
+        query,
+        fuzziness: options.fuzziness,
+        filter: options.filter as Schemas.FilterExpression
```

Comment:
I need to battle with these types. Opening https://github.com/xataio/client-ts/issues/313